src/Maybe/README.md

```diff
@@ -0,0 +1,1321 @@
+const chain = require('crocks/pointfree/chain')
+const compose = require('crocks/helpers/compose')
+const ifElse = require('crocks/logic/ifElse')
+const isNumber = require('crocks/predicates/isNumber')
+
+// gt5 :: Number -> Boolean
+const gt5 =
+  x => x > 5
+
+// safe :: (a -> Boolean) -> a -> Maybe b
+const safe = pred =>
+  ifElse(pred, Just, Nothing)
+
+// safeNumber :: a -> Maybe Number
+const safeNumber =
+  safe(isNumber)
+
+// maybeBig :: Number -> Maybe Number
+const maybeBig =
+  safe(gt5)
+
+// bigNumber :: a -> Maybe Number
+const bigNumber = compose(
+  chain(maybeBig), safeNumber
+)
+
+safeNumber(45)
+//=> Just 45
+
+safeNumber('99')
+//=> Nothing
+
+maybeBig(99)
+//=> Just 99
+
+maybeBig(2)
+//=> Nothing
+
+bigNumber(34)
+//=> Just 34
+
+bigNumber('string')
+//=> Nothing
+
+bigNumber(3)
+//=> Nothing
+```
+
+## Implements
+`Setoid`, `Semigroup`, `Functor`, `Alt`, `Plus`, `Apply`, `Traversable`,
+`Chain`, `Applicative`, `Alternative`, `Monad`
+
+## Constructor Methods
+
+#### Nothing
+
+```haskell
+Maybe.Nothing :: () -> Maybe a
+```
+
+Used to construct a `Nothing` instance that represents the "false" portion of
+a disjunction. When an instance is a `Nothing`, most `Maybe` returning methods
+will just return another `Nothing`. Anything passed to the constructor will
+be thrown out and mapped to `()`.
+
+```javascript
+const { Just, Nothing } = require('crocks/Maybe')
+
+const chain = require('crocks/pointfree/chain')
+const isNumber = require('crocks/predicates/isNumber')
+const safeLift = require('crocks/Maybe/safeLift')
+
+// add10 :: Number -> Number
+const add10 =
+  x => x + 10
+
+// safeAdd10 :: a -> Maybe Number
+const safeAdd10 =
+  safeLift(isNumber, add10)
+
+Just(23)
+  .map(add10)
+//=> Just 33
+
+Nothing(23)
+  .map(add10)
+//=> Nothing
+
+chain(safeAdd10, Just(10))
+//=> Just 20
+
+chain(safeAdd10, Nothing())
+//=> Nothing
+```
+
+#### Just
+
+```haskell
+Maybe.Just :: a -> Maybe a
+```
+
+Used to construct a `Just` instance that represents the "true" portion of a
+disjunction or a valid value. `Just` will wrap any given value in a `Just`,
+which signals the validity of the wrapped value.
+
+```javascript
+const { Just, Nothing } = require('crocks/Maybe')
+
+const compose = require('crocks/helpers/compose')
+const ifElse = require('crocks/logic/ifElse')
+const isString = require('crocks/predicates/isString')
+const map = require('crocks/pointfree/map')
+
+// toUpper :: String -> String
+const toUpper =
+  x => x.toUpperCase()
+
+// safe :: (a -> Boolean) -> a -> Maybe a
+const safe =
+  pred => ifElse(pred, Just, Nothing)
+
+// safeShout :: a -> Maybe String
+const safeShout = compose(
+  map(toUpper),
+  safe(isString)
+)
+
+safeShout(45)
+//=> Nothing
+
+safeShout('Hey there!')
+//=> Just "HEY THERE!"
+```
+
+#### of
+
+```haskell
+Maybe.of :: a -> Maybe a
+```
+
+Used to wrap any value into a `Maybe` as a `Just`, `of` is used mostly by
+helper functions that work "generically" with `Applicative` or `Monad` types
```

Comment:
"types" should probs be "type" as "instances" covers the plurality?

Reply:
Actually think we can :hocho: the word `type` altogether.